src/osd/ReplicatedPG.cc

```diff
@@ -1800,6 +1800,10 @@ bool ReplicatedPG::maybe_handle_cache(OpRequestRef op,
     osd->logger->inc(l_osd_op_cache_hit);
     return false;
   }
+  
+  if (op->need_skip_promote()) {
+    return false;
+  }
```

Comment:
actually it is probably a bit safer to move this below the msut_promote case below.  I don't think they will conflict right now unless there is a notify + delete combined (whcih doesn't make sense), but we should avoid potential issues later.